pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -686,6 +687,42 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	return
 }
 
+func (m *kubeGenericRuntimeManager) preStopSidecars(pod *v1.Pod, containers []*kubecontainer.Container) {
```

Comment:
I'd suggest we test this function, at least for the part that determines container grace period and whether we should call `executePreStopHook `. it's ok we add more helpers